Branch-TestBed/Branch-TestBed/AppDelegate.m

```diff
@@ -52,10 +62,45 @@ - (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(
             NSLog(@"Branch TestBed: Initialization failed\n%@", error.localizedDescription);
         }
     }];
-    
+
     return YES;
 }
 
+- (void)onboardUserOnInstall {
+    NSURL *urlForOnboarding = [NSURL URLWithString:@"http://example.com"]; // Put your onboarding link here
+    
+    id notInstall = [[NSUserDefaults standardUserDefaults] objectForKey:@"notInstall"];
+    if (!notInstall) {
+        [[NSUserDefaults standardUserDefaults] setObject:@YES forKey:@"notInstall"];
+        [[NSUserDefaults standardUserDefaults] synchronize];
+
+        Branch *branch = [Branch getInstance];
+
+        // Note that this must be invoked *before* initSession
+        [branch enableDelayedInit];
+
+        NSURL *updatedUrlForOnboarding = [branch getUrlForOnboardingWithRedirectUrl:urlForOnboarding.absoluteString];
+        if (updatedUrlForOnboarding) {
+            // replace url for onboarding with the URL provided by Branch
+            urlForOnboarding = updatedUrlForOnboarding;
+        }
+        else {
+            // do not replace url for onboarding
+            NSLog(@"Was unable to get onboarding URL from Branch SDK, so proceeding with normal onboarding URL.");
+            [branch disableDelayedInit];
+        }
+
+        self.onboardingVC = [[SFSafariViewController alloc] initWithURL:urlForOnboarding];
+        self.onboardingVC.delegate = self;
+        dispatch_async(dispatch_get_main_queue(), ^{
+            [[[[UIApplication sharedApplication].delegate window] rootViewController] presentViewController:self.onboardingVC animated:YES completion:NULL];
+        });
+    }
+}
+
+- (void)safariViewControllerDidFinish:(SFSafariViewController *)controller {
```

Comment:
I thought about this yesterday when coding this up and realized that would be problematic. initSession would finish and we'd fire the callback with the SFSafariViewController onscreen. If deep link data comes back, then potentially we'd be trying to push a view controller while the SFSafariViewController is still on screen.

The downside of this example is that init is delayed until after the VC is dismissed when optimally this would occur while the VC is still onscreen. However, for the sake of simplicity, I think we should keep it like this. If partners want, they can implement a more complex setup that proceeds with init while the VC is still onscreen, and add more logic to the initSession callback that delays pushing any new VCs until after SFSafariViewController is dismissed. Given that everyone's view controller hierarchy is unique, I'd rather keep the example here as agnostic as possible to anyone's particular setup. Is that okay?